### PR TITLE
Extend IPlacedComponent with world position/orientation getters.

### PR DESCRIPTION
**Pull request by poirierlouis**
Provide getters on scripting side since RE of property with https://github.com/WopsS/RED4ext.SDK/pull/138.

I tested the code in my own RED4ext plugin up until now, which is working fine. But I haven't tried with Codeware (workspace not configured to build plugin/scripts and test it yet).

### CODE REVIEW

File: src/App/Entity/PlacedComponentEx.hpp [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+namespace App
+{
+struct PlacedComponentEx : Red::IPlacedComponent
+{
+    Vector4 GetWorldPosition() const;
+    Quaternion GetWorldOrientation() const;
+};
+}
+
+RTTI_EXPAND_CLASS(Red::IPlacedComponent, App::PlacedComponentEx, {
```
Comment (psiberx) [inline src/App/Entity/PlacedComponentEx.hpp:12]: I'd register the property itself.

**Review by psiberx — CHANGES_REQUESTED** [review]

**Comment by poirierlouis** [comment]
Done :)
Is [README](https://github.com/psiberx/cp2077-red-lib/?tab=readme-ov-file#class-extensions) of RedLib obsolete regarding extension of class properties? Or I'm missing something in this case?

File: src/App/Entity/PlacedComponentEx.hpp [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#pragma once
+
+namespace App
+{
+struct PlacedComponentEx : Red::IPlacedComponent
```
Comment (psiberx) [inline src/App/Entity/PlacedComponentEx.hpp:5]: You don't need extra struct if you just want to register a property. Example: https://github.com/psiberx/cp2077-codeware/blob/main/src/App/Depot/MeshEx.hpp

**Review by psiberx — CHANGES_REQUESTED** [review]

**Comment by psiberx** [comment]
It doesn't have an example for properties.

**Comment by poirierlouis** [comment]
My bad, should be good now.

> Properties cannot be added to existing classes.

I was talking about this note. Which I guess is an old statement?

**Comment by psiberx** [comment]
You can't add properties to a C++ class, but you can register more properties in RTTI.

**Comment by poirierlouis** [comment]
Alright, well the later was my expectation when using RTTI_PROPERTY, but the note was not explicit to me, I thought it wasn't possible to declare more RTTI properties when extending. I'll write a PR about this.

**Pull request merged**